packages/dds/tree/src/tableSchema.ts

```diff
@@ -768,31 +759,31 @@ export namespace System_TableSchema {
 						return [];
 					}
 
+					// Ensure the specified columns exists before starting transaction.
+					// Improves user-facing error experience.
+					for (const columnToRemove of columnsToRemove) {
+						this._assertContainsColumn(columnToRemove);
+					}
+
 					const removedColumns: ColumnValueType[] = [];
-					Tree.runTransaction(this, () => {
+					this._applyEditsInBatch(() => {
 						// Note, throwing an error within a transaction will abort the entire transaction.
 						// So if we throw an error here for any column, no columns will be removed.
 						for (const columnToRemove of columnsToRemove) {
-							const removedColumn = this.removeColumn(columnToRemove);
+							const removedColumn = this._removeColumn(columnToRemove);
```

Comment:
Note for reviewers: this code previously had a mix of private methods that did and did not include the `_` prefix. Since it is needed for a couple of methods to avoid conflicting with the public API, I went ahead and just added it to all of the private methods.